waku/waku_rln_relay/group_manager/static/group_manager.nim

```diff
@@ -44,51 +44,90 @@ method startGroupSync*(g: StaticGroupManager): Future[void] {.async: (raises: [E
   initializedGuard(g)
   # No-op
 
-method register*(g: StaticGroupManager, idCommitment: IDCommitment):
-                 Future[void] {.async: (raises: [Exception]).} =
-  initializedGuard(g)
-
-  await g.registerBatch(@[idCommitment])
-
-
-method registerBatch*(g: StaticGroupManager, idCommitments: seq[IDCommitment]):
-                      Future[void] {.async: (raises: [Exception]).} =
-  initializedGuard(g)
-
-  let membersInserted = g.rlnInstance.insertMembers(g.latestIndex + 1, idCommitments)
-  if not membersInserted:
-    raise newException(ValueError, "Failed to insert members into the merkle tree")
-
-  if g.registerCb.isSome():
-    var memberSeq = newSeq[Membership]()
-    for i in 0..<idCommitments.len():
-      memberSeq.add(Membership(idCommitment: idCommitments[i], index: g.latestIndex + MembershipIndex(i) + 1))
-    await g.registerCb.get()(memberSeq)
-
-  discard g.slideRootQueue()
-
-  g.latestIndex += MembershipIndex(idCommitments.len())
-
-  return
-
-method withdraw*(g: StaticGroupManager, idSecretHash: IdentitySecretHash):
-                 Future[void] {.async: (raises: [Exception]).} =
-  initializedGuard(g)
-
-  let groupKeys = g.groupKeys
-
-  for i in 0..<groupKeys.len():
-    if groupKeys[i].idSecretHash == idSecretHash:
-        let idCommitment = groupKeys[i].idCommitment
-        let index = MembershipIndex(i)
-        let memberRemoved = g.rlnInstance.removeMember(index)
-        if not memberRemoved:
-          raise newException(ValueError, "Failed to remove member from the merkle tree")
-
-        if g.withdrawCb.isSome():
-          await g.withdrawCb.get()(@[Membership(idCommitment: idCommitment, index: index)])
-
-        return
+when defined(rln_v2):
+  method register*(g: StaticGroupManager, 
+                   rateCommitment: RateCommitment): Future[void] {.async: (raises: [Exception]).} =
+    initializedGuard(g)
+
+    await g.registerBatch(@[rateCommitment])
+
+else:
+  method register*(g: StaticGroupManager, idCommitment: IDCommitment):
+                  Future[void] {.async: (raises: [Exception]).} =
+    initializedGuard(g)
+
+    await g.registerBatch(@[idCommitment])
+
+when defined(rln_v2):
+  method registerBatch*(g: StaticGroupManager, 
+                        rateCommitments: seq[RateCommitment]): Future[void] {.async: (raises: [Exception]).} =
+    initializedGuard(g)
+
+    let leavesRes = rateCommitments.toLeaves()
+    if not leavesRes.isOk():
+      raise newException(ValueError, "Failed to convert rate commitments to leaves")
+    let leaves = cast[seq[seq[byte]]](leavesRes.get())
```

Comment:
May it be more explicit?
```suggestion
    let leaves = cast[seq[IDCommitment]](leavesRes.get())
```

Reply:
this may cause confusion about converting the rateCommitment into an IDCommitment, when that is not the case. here we have used seq[byte] to discern it from an IDCommitment. I don't see the value in creating a type for it i.e SerializedRateCommitment yet.